newt/install/install.go

```diff
@@ -647,17 +647,29 @@ func (inst *Installer) calcVersionMap(candidates []*repo.Repo) (
 	}
 
 	// If project.yml specified any specific git commits, ensure we get them.
+	rg := dg.Reverse()
 	for name, ver := range vm {
-		reqs := inst.reqs[name]
-		if len(reqs) > 0 {
-			keep, err := inst.shouldKeepCommit(name, reqs[0].Ver.Commit)
-			if err != nil {
-				return nil, err
-			}
-			if keep {
-				ver.Commit = reqs[0].Ver.Commit
+		for _, node := range rg[name] {
+			if len(node.VerReqs) > 0 {
+				// look at all elements of VerReqs array
```

Comment:
This is 100% a nit pick so feel free to ignore it.  I find comments like this one don't provide much signal vs. noise.  I.e., we can tell from the next line that we are looping over the `VerReqs` slice, so no need to say it twice.